core/shared/src/main/scala/org/http4s/util/SizedSeq.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.util
+
+sealed trait SizedSeq[+A] {
+  def toSeq: Seq[A]
+}
+
+sealed abstract class SizedSeqBase[+A] extends SizedSeq[A] {
```

Comment:
I'm not sure what this intermediate type achieves.  Can't the concrete ones just override `toSeq` directly with a val?

Reply:
Didn't know I can override a method with `val`, thanks. Made the change.